File: android-core/src/test/java/com/mparticle/networking/CertificateTest.java
package com.mparticle.networking;

import com.mparticle.testutils.RandomUtils;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

public class CertificateTest {
    RandomUtils randomUtils = new RandomUtils();
    String alias = randomUtils.getAlphaNumericString(10);
    String certificateString = randomUtils.getAlphaNumericString(124);

    @Test
    public void buildCertificateTest() {
        Certificate certificate = Certificate.with(alias, certificateString);
        assertEquals(alias, certificate.getAlias());
        assertEquals(certificateString, certificate.getCertificate());
    }

    @Test
    public void rejectMalformedCertificate() {
        Certificate certificate = Certificate.with("", certificateString);
        assertNull(certificate);

        certificate = Certificate.with(alias, "");
        assertNull(certificate);

    }
}

File: android-core/src/test/kotlin/com/mparticle/networking/CertificateTest.kt
package com.mparticle.networking

import com.mparticle.testutils.RandomUtils
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test

class CertificateTest {
    val randomUtils = RandomUtils()
    val alias = randomUtils.getAlphaNumericString(10)
    val certificateString = randomUtils.getAlphaNumericString(124)

    @Test
    fun buildCertificateTest() {
        val certificate = Certificate.with(alias, certificateString)
        assertEquals(alias, certificate?.getAlias())
        assertEquals(certificateString, certificate?.getCertificate())
    }

    @Test
    fun rejectMalformedCertificate() {
        var certificate = Certificate.with("", certificateString)
        assertNull(certificate)

        certificate = Certificate.with(alias, "")
        assertNull(certificate)
    }
}
